### ISSUE TITLE
Telescope does not make use of secure_asset

### ISSUE THREAD

**Issue by rennokki** [laravel/telescope#556]
I am currently using K8s with Laravel and i am somehow forced to use secure_asset() instead of asset() everywhere. Telescope does not use secure_asset() and this causes a huge break on design when accessing telescope.

**Comment by GigaMick** [comment]
Are you seeing HUGE svgs?

**Comment by rennokki** [comment]
Yes. It's the only thing i see and it cannot be unseen.

**Comment by Cheezykins** [comment]
Yeah we use a reverse proxy to access telescope which does the edge termination of SSL, so telescope serves assets using the wrong scheme.

**Comment by rennokki** [comment]
I had nothing to choose but to use a reverse proxy too.  💃

**Comment by rolandstarke** [comment]
Would adding this code inside your `AppServiceProvider`s `boot` method help?

```php
        if(config('app.env') === 'production') {
            \URL::forceScheme('https');
        }
```

source: https://stackoverflow.com/a/45731506/5378743




Hardcoding `secure_asset` everywhere would make it impossible to run over http.

**Comment by rennokki** [comment]
I tried that. It seems good 🗡
Also, i'd use it this way:

```php
if ($this->app->environment(['staging', 'production'])) {
    URL::forceScheme('https');
}
```

**Comment by Cheezykins** [comment]
Laravel itself handles this using the [fideloper/proxy](https://github.com/fideloper/TrustedProxy) package and adhering to the X-FORWARDED-SCHEME header - however having it set for telescope has no effect.

Could telescope be modified to also use this package considering it's a [hard dependency of Laravel already](https://github.com/laravel/laravel/blob/2a1f3761e89df690190e9f50a6b4ac5ebb8b35a3/composer.json#L12)?

**Comment by rennokki** [comment]
Might be the next best thing, but not quite me. Solved using forced schemas. I hope Google indexed this issue properly so people searching on Google the same kind of problem would get the answer. 👏

**Comment by amribrahim34** [comment]
I faced the same issue and the solution was to use the ASSET_URL environmental variable and paste the https version of my url

**Comment by dir** [comment]
Can confirm that the `ASSET_URL` fix resolved the HUGE SVG of an eye staring me in my soul as I contemplated why I couldn't get Telescope to work on my deployments. Thank you @amribrahim34 

This is what I did in my `config/app.php`:

`'asset_url' => env('APP_URL'),`

I already use HTTPS as the app url on my deployments so that's all I needed. I find it strange that I've never noticed this random config var before, and that is has no default value.
